### ISSUE TITLE
azblob v0.5.0 and blockBlobClient path joining behaviour

### ISSUE THREAD

**Issue by vijaytdh** [Azure/azure-sdk-for-go#19245]
### Bug Report

- import path of package in question
```
	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blockblob"
	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
```
- SDK version: `v0.5.0` of `sdk/storage/azblob`
- output of `go version`:   ` go1.18.2 linux/amd64`

- What happened?

Given code similar to:

```
package main

import (
        "context"
        "fmt"

        "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
        "github.com/sirupsen/logrus"
)

func main() {
        ctx := context.Background()
        token := "?some token goes here"
        containerURL := fmt.Sprintf("https://mystorage.blob.core.windows.net/mycontainer%s", token)
        containerClient, err := container.NewClientWithNoCredential(containerURL, nil)
        if err != nil {
                logrus.Fatal(err)
        }

        blobURL := "test.txt"
        blobClient := containerClient.NewBlockBlobClient(blobURL)
        props, err := blobClient.GetProperties(ctx, nil)
        if err != nil {
                logrus.Fatal(err)
        }
        fmt.Println(props.ContentMD5)
}
```

Results in the error:

```
FATA[0000] HEAD https://mystorage.blob.core.windows.net/mycontainer/test.txt/
--------------------------------------------------------------------------------
RESPONSE 404: 404 The specified blob does not exist.
ERROR CODE: BlobNotFound
--------------------------------------------------------------------------------
Response contained no body
```

Note the extra forward slash "/" at the end. This is because when instantiating the client the code calls runtime.JoinPaths [here](https://github.com/Azure/azure-sdk-for-go/blob/sdk/storage/azblob/v0.5.0/sdk/storage/azblob/container/client.go#L124). 

 Which is this code [here](https://github.com/Azure/azure-sdk-for-go/blob/sdk/azcore/v1.1.3/sdk/azcore/runtime/request.go#L48).

- What did you expect or want to happen?

I was expecting I would be able to create a container client and a blockblock client, passing in the blob name and that would work but it results in this extra slash. Is this intended behaviour or am I not supposed to create the blockblob client in this way?
Should I be using the blockblob package and instantiating a blockblob.NewClientWithNoCredential instead?

It is worth noting that if I use `azidentity.NewDefaultAzureCredential` and `container.NewClient` passing in the credential - then I do not encounter this issue. Which makes me think `containerClient.NewBlockBlobClient` is only intended to be used when passing in an AAD credential. So perhaps this isn't a bug but intended behaviour. However, it would be good if there wasn't this difference between using SAS tokens and AAD tokens.

- How can we reproduce it?
Create a storage account, container and SAS token. Replace the values in the code and run it.

- Anything we should know about your environment.
N/A

**Comment by stijndehaes** [comment]
I am seeing the same issue when using an SAS URL, was in the process of upgrading from an older version but this breaks things

**Comment by drakkan** [comment]
This is an issue for my use case as well, a fix would be appreciated, thanks

**Comment by jhendrixMSFT** [comment]
The fix is in `azcore@v1.1.4` which is scheduled for release on 10/6/2022.
